drkafka/src/main/java/com/pinterest/doctorkafka/KafkaCluster.java

```diff
@@ -71,36 +68,31 @@ public String name() {
    *
    * @param brokerStats  the broker stats
    */
-  public synchronized void recordBrokerStats(BrokerStats brokerStats) {
+  public void recordBrokerStats(BrokerStats brokerStats) {
     try {
       int brokerId = brokerStats.getId();
-      if (!brokerStatsMap.containsKey(brokerId)) {
-        brokerStatsMap.put(brokerId, new LinkedList<>());
-      }
-      LinkedList<BrokerStats> brokerStatsList = brokerStatsMap.get(brokerId);
-      if (brokerStatsList.size() == MAX_NUM_STATS) {
-        brokerStatsList.removeFirst();
+      LinkedList<BrokerStats> brokerStatsList = brokerStatsMap.computeIfAbsent(brokerId, i -> new LinkedList<>());
+
+      synchronized (brokerStatsList){
```

Comment:
Could you add comments to explain why we need mutual exclusive access to `brokerStatsList` here?   what are the concurrent reader / writers to this shared variable?